drivers/storage/portworx/component/disruption_budget.go

```diff
@@ -100,31 +100,41 @@ func (c *disruptionBudget) createPortworxPodDisruptionBudget(
 	cluster *corev1.StorageCluster,
 	ownerRef *metav1.OwnerReference,
 ) error {
-	var err error
-	c.sdkConn, err = pxutil.GetPortworxConn(c.sdkConn, c.k8sClient, cluster.Namespace)
+	userProvidedMinValue, err := pxutil.MinAvailableForStoragePDB(cluster)
 	if err != nil {
-		return err
+		logrus.Warnf("Invalid value for annotation %s: %v", pxutil.AnnotationStoragePodDisruptionBudget, err)
 	}
 
-	storageNodesCount, err := pxutil.CountStorageNodes(cluster, c.sdkConn, c.k8sClient)
-	if err != nil {
-		c.closeSdkConn()
-		return err
+	var minAvailable int
+	if err != nil || userProvidedMinValue < 0 {
+		c.sdkConn, err = pxutil.GetPortworxConn(c.sdkConn, c.k8sClient, cluster.Namespace)
+		if err != nil {
+			return err
+		}
+
+		storageNodesCount, err := pxutil.CountStorageNodes(cluster, c.sdkConn, c.k8sClient)
+		if err != nil {
+			c.closeSdkConn()
```

Comment:
Since we called `pxutil.GetPortworxConn()` 7 lines up, shouldn't we be calling the `defer c.closeSdkConn()` unconditionally?

Reply:
We don't close the connection on happy path as we go through this every 30s. We close it on failure because the failure could be because of the connection.
The `GetPortworxConn` method returns the same `sdkConn` object it it already exists and doesn't create a new one.